aspnetcore/blazor/state-management.md

```diff
@@ -114,46 +201,273 @@ Caveats for using browser storage:
+}
+
+@code {
+    private bool isLoaded;
+
+    [Parameter]
+    public RenderFragment ChildContent { get; set; }
+
+    public int CurrentCount { get; set; }
+
+    protected override async Task OnInitializedAsync()
+    {
+        var result = await ProtectedSessionStore.GetAsync<int>("count");
+        currentCount = result.Success ? result.Value : 0;
+        isLoaded = true;
+    }
+
+    public async Task SaveChangesAsync()
+    {
+        await ProtectedSessionStore.SetAsync("count", CurrentCount);
+    }
+}
+```
+
+The `CounterStateProvider` component handles the loading phase by not rendering its child content until loading is complete.
+
+To use the `CounterStateProvider` component, wrap an instance of the component around any other component that requires access to the counter state. To make the state accessible to all components in an app, wrap the `CounterStateProvider` component around the <xref:Microsoft.AspNetCore.Components.Routing.Router> in the `App` component (`App.razor`):
+
+```razor
+<CounterStateProvider>
+    <Router AppAssembly="typeof(Startup).Assembly">
+        ...
+    </Router>
+</CounterStateProvider>
+```
+
+Wrapped components receive and can modify the persisted counter state. The following `Counter` component implements the pattern:
+
+```razor
+@page "/counter"
+
+<p>Current count: <strong>@CounterStateProvider.CurrentCount</strong></p>
+<button @onclick="IncrementCount">Increment</button>
+
+@code {
+    [CascadingParameter]
+    private CounterStateProvider CounterStateProvider { get; set; }
+
+    private async Task IncrementCount()
+    {
+        CounterStateProvider.CurrentCount++;
+        await CounterStateProvider.SaveChangesAsync();
+    }
+}
+```
+
+The preceding component isn't required to interact with `ProtectedBrowserStorage`, nor does it deal with a "loading" phase.
+
+To deal with prerendering as described earlier, `CounterStateProvider` can be amended so that all of the components that consume the counter data automatically work with prerendering. For more information, see the [Handle prerendering](#handle-prerendering) section.
+
+In general, the *state provider parent component* pattern is recommended:
+
+* To consume state across many components.
+* If there's just one top-level state object to persist.
+
+To persist many different state objects and consume different subsets of objects in different places, it's better to avoid persisting state globally.
+
+::: moniker-end
+
+::: moniker range="< aspnetcore-5.0"
+
+## Protected Browser Storage experimental NuGet package
+
+The [`Microsoft.AspNetCore.ProtectedBrowserStorage`](https://www.nuget.org/packages/Microsoft.AspNetCore.ProtectedBrowserStorage) experimental NuGet package leverages [ASP.NET Core Data Protection](xref:security/data-protection/introduction) for [`localStorage`](https://developer.mozilla.org/docs/Web/API/Window/localStorage) and [`sessionStorage`](https://developer.mozilla.org/docs/Web/API/Window/sessionStorage).
```

Comment:
No need to mention it's experimental because you call that out in the warning message.

```suggestion
The [`Microsoft.AspNetCore.ProtectedBrowserStorage`](https://www.nuget.org/packages/Microsoft.AspNetCore.ProtectedBrowserStorage) NuGet package leverages [ASP.NET Core Data Protection](xref:security/data-protection/introduction) for [`localStorage`](https://developer.mozilla.org/docs/Web/API/Window/localStorage) and [`sessionStorage`](https://developer.mozilla.org/docs/Web/API/Window/sessionStorage).
```